src/plugins/walletConnectToDapps/components/WalletConnectSigningModal/content/TransactionSimulation.tsx

```diff
@@ -33,12 +34,21 @@ export const TransactionSimulation: FC<TransactionSimulationProps> = ({ transact
 
   const { simulationQuery } = useSimulateEvmTransaction({ transaction, chainId, speed })
 
+  const [cachedSimulationData, setCachedSimulationData] =
```

Comment:
Once we have a sim, we have a sim - since this is a component instance for a given `transaction`, then effectively simulated data never changes as it's deterministically the same regardless of speed (gas params).